utils/stats.py

```diff
@@ -147,8 +154,30 @@ def stats_card(
         # move the cursor to the beginning
         image_binary.seek(0)
 
-        file = discord.File(fp=image_binary, filename=f"{leetcode_id}.png")
+        file = discord.File(fp=image_binary, filename=f"{filename}.png")
 
     os.remove(paths[0])
 
     return file
+
+def anonymise_stats_card(path: str) -> None:
+    """
+    Anonymise the stats card at the given path using Pillow (PIL).
+
+    This function modifies any given stats card to remove the user's
+    LeetCode ID from the image if they want to be anonymous.
+
+    :param path: The path to the stats card image.
+    """
+    try:
+        stats_card = Image.open(path)
+
+        hidden_banner = Image.open("ui\\hidden_banner.png")
+
+        region = hidden_banner.crop((0, 0, 435, 30))
+
+        stats_card.paste(region, (60, 20, 495, 50))
+
+        stats_card.save(path)
+    except Exception as e:
+        print(f"An error occurred while anonymising the stats card: {e}")
```

Comment:
It is good practice to end files with a newline, and that convention is followed in this project (although automatically due to the black formatter used)